ocs_ci/ocs/platform_nodes.py

```diff
@@ -432,3 +467,424 @@ def restart_nodes_teardown(self):
+                                )
+
+    def get_ready_status(self, node_ent):
+        """
+        Get the node 'Ready' status
+
+        Args:
+            node_ent (dict): Node info which includes details
+
+        Returns:
+            bool: True if node is Ready else False
+
+        """
+        for cond in node_ent['status']['conditions']:
+            if cond['type'] == 'Ready':
+                if not cond['status'] == "True":
+                    return False
+                else:
+                    return True
+
+    def build_ansible_inventory(self, hosts):
+        """
+        Build the ansible hosts file from jinja template
+
+        Args:
+            hosts (list): list of private host names
+
+        Returns:
+            str: path of the ansible file created
+
+        """
+        _templating = templating.Templating()
+        ansible_host_file = dict()
+        ansible_host_file['ansible_user'] = constants.EC2_USER
+        ansible_host_file['ansible_become'] = 'True'
+        ansible_host_file['pod_kubeconfig'] = '/kubeconfig'
+        ansible_host_file['pod_pull_secret'] = '/tmp/pull-secret'
+        ansible_host_file['rhel_worker_nodes'] = hosts
+
+        logging.info(ansible_host_file)
+        data = _templating.render_template(
+            constants.ANSIBLE_INVENTORY_YAML,
+            ansible_host_file,
+        )
+        logging.debug("Ansible hosts file:%s", data)
+        host_file_path = "/tmp/hosts"
+        with open(host_file_path, 'w') as f:
+            f.write(data)
+        return host_file_path
+
+    @retry(exceptions.CommandFailed, tries=15, delay=30, backoff=1)
+    def check_connection(self, rhel_pod_obj, host, pem_dst_path):
+        """
+        Check whether newly brought up RHEL instances are accepting
+        ssh connections
+
+        Args:
+            rhel_pod_obj (Pod): object for handling ansible pod
+            host (str): Node to which we want to try ssh
+            pem_dst_path (str): path to private key for ssh
+
+        """
+        cmd = 'ls'
+        rhel_pod_obj.exec_cmd_on_node(
+            host, pem_dst_path, cmd, user=constants.EC2_USER
+        )
+
+
+class AWSUPINode(AWSNodes):
+    """
+    Node object representing AWS upi nodes
+
+    """
+    def __init__(self, node_conf, node_type):
+        super(AWSUPINode, self).__init__()
+        self.node_conf = node_conf
+        #  RHEL/RHCOS
+        self.node_type = node_type
+        #  This variable will hold the AWS instance object
+        self.aws_instance_obj = None
+        self.region = config.ENV_DATA['region']
+        self.cluster_name = get_cluster_name(self.cluster_path)
+        self.client = boto3.client(
+            'ec2', region_name=self.region
+        )
+        # cloudformation
+        self.cf = boto3.client('cloudformation', region_name=self.region)
+
+    def _prepare_node(self):
+        """
+        Create AWS instance of the node
+
+        """
+        if self.node_type == 'RHEL':
```

Comment:
how are we handling RHCOS

Reply:
I will add RHCOS function once this PR gets merged. I was comfortable with RHEL steps hence sticking to this .